util/ivmFactory.js

```diff
@@ -46,10 +46,12 @@ async function createIvm(code, libraryVersionIds) {
           outputEvents = transformBatch(eventMessages, metadata).map(transformedEvent => ({transformedEvent, metadata: metadata(transformedEvent)}))
           break;
         case "transformEvent":
-          outputEvents = eventMessages.map(ev => {
-            let transformedEvent = transformEvent(ev, metadata)
+
+          outputEvents = await Promise.all(eventMessages.map(async ev => {
```

Comment:
As @gane5hvarma pointed out in an earlier conversation, this will mess up the event ordering if we run transformEvent in an async fashion.

Reply:
With promise.all, returned values will be in the order of the Promises passed, regardless of the completion order.

Reply:
But, if any of the promises is rejected, it rejects the whole array. I will change the logic here.
Also will add more test cases around when some of the promises getting rejected.

Reply:
@lokey so if any of the promise is rejected, should we drop whole batch and raise an error ?